### PR TITLE
ncm-metaconfig: hadoop config

### PR DESCRIPTION
**Pull request by kwaegema**
added a metaconfig service for hadoop config files with some basic schema

### CODE REVIEW

**Comment by jrha** [comment]
Nice, I'll try to take a look at this.

File: ncm-metaconfig/src/main/metaconfig/hadoop/tests/profiles/config.pan [hunk 1]
```diff
@@ -0,0 +1,48 @@
+object template config;
+
+include 'metaconfig/hadoop/config';
+
+
+bind "/software/components/metaconfig/services/{/etc/hadoop/conf.quattor/core-site.xml}/contents" = hadoop_core_site;
+bind "/software/components/metaconfig/services/{/etc/hadoop/conf.quattor/hdfs-site.xml}/contents/dfs" = hadoop_hdfs_site;
+#bind "/software/components/metaconfig/services/{/etc/hadoop/conf.quattor/slaves}/contents" = type_hdfs_slaves;
```
Comment (stdweird) [inline ncm-metaconfig/src/main/metaconfig/hadoop/tests/profiles/config.pan]: remove if unused

File: ncm-metaconfig/src/main/metaconfig/hadoop/tests/profiles/config.pan [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+prefix "/software/components/metaconfig/services/{/etc/hadoop/conf.quattor/hdfs-site.xml}";
+"module" = "hadoop/main";
+prefix "/software/components/metaconfig/services/{/etc/hadoop/conf.quattor/core-site.xml}";
+"module" = "hadoop/main";
+#prefix "/software/components/metaconfig/services/{/etc/hadoop/conf.quattor/slaves}";
```
Comment (stdweird) [inline ncm-metaconfig/src/main/metaconfig/hadoop/tests/profiles/config.pan]: remove if unused

File: ncm-metaconfig/src/main/metaconfig/hadoop/tests/profiles/config.pan [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+'namenode/handler.count' = 400;
+prefix "/software/components/metaconfig/services/{/etc/hadoop/conf.quattor/hdfs-site.xml}/contents/dfs/datanode";
+'handler.count' = 40;
+'max.transfer.threads' = 8192;
+'address'  = 'localhost:50010';
+'ipc.address' ='localhost:50020';
+'http.address' = 'localhost:50075';
+'https.address' = 'localhost:50475';
+
+#"/software/components/metaconfig/services/{/etc/hadoop/conf.quattor/slaves}/contents" = list('localhost', 'remotehost');
```
Comment (stdweird) [inline ncm-metaconfig/src/main/metaconfig/hadoop/tests/profiles/config.pan]: remove if unused

**Comment by stdweird** [comment]
@kwaegema minoro remarks, LGTM otherwise

**Comment by kwaegema** [comment]
@stdweird done

**Comment by stdweird** [comment]
LGTM

File: ncm-metaconfig/src/main/metaconfig/hadoop/pan/schema.pan [hunk 4]
```diff
@@ -0,0 +1,52 @@
+declaration template metaconfig/hadoop/schema;
+
+include 'pan/types';
+
+type hadoop_core_site_fs_defaultFS  = {
```
Comment (piojo) [inline ncm-metaconfig/src/main/metaconfig/hadoop/pan/schema.pan]: Annotations, please :smile:

File: ncm-metaconfig/src/main/metaconfig/hadoop/tests/profiles/config.pan [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+include 'metaconfig/hadoop/config';
+
+
+bind "/software/components/metaconfig/services/{/etc/hadoop/conf.quattor/core-site.xml}/contents" = hadoop_core_site;
+bind "/software/components/metaconfig/services/{/etc/hadoop/conf.quattor/hdfs-site.xml}/contents/dfs" = hadoop_hdfs_site;
+bind "/software/components/metaconfig/services/{/etc/hadoop/conf.quattor/slaves}/contents" = type_hdfs_slaves;
+
+prefix "/software/components/metaconfig/services/{/etc/hadoop/conf.quattor/hdfs-site.xml}";
+"module" = "hadoop/main";
```
Comment (piojo) [inline ncm-metaconfig/src/main/metaconfig/hadoop/tests/profiles/config.pan]: Save yourself the `prefix` mess:

```bash
"/software/components/metaconfig/services/{/etc/hadoop/conf.quattor/hdfs-site.xml}/module" = "hadoop/main";
"/software/components/metaconfig/services/{/etc/hadoop/conf.quattor/core-site.xml}/module = "hadoop/main";
...
```

Or,

```bash
prefix "/software/components/metaconfig/services";
"{/etc/hadoop/conf.quattor/core-site.xml}/module" = "hadoop/main";
"{/etc/hadoop/conf.quattor/hdfs-site.xml}/module" = "hadoop/main";
"{/etc/hadoop/conf.quattor/slaves}/module" = "hadoop/main";
```

**Comment by kwaegema** [comment]
@piojo done!

**Pull request merged**